File: resource-server/src/main/java/com/elearning/course/application/dto/CourseDTO.java
package com.elearning.course.application.dto;

import com.elearning.course.domain.Course;
import com.elearning.course.domain.Language;
import jakarta.validation.constraints.NotBlank;

import java.util.Set;

public record CourseDTO (
        @NotBlank(message = "Title is required")
        String title,
        String description,
        String thumbnailUrl,
        Set<String> benefits,
        Language language,
        Set<String> prerequisites,
        Set<Language> subtitles
) {

    public Course toCourse(String teacher) {
        return new Course(
                title,
                description,
                thumbnailUrl,
                benefits,
                language,
                prerequisites,
                subtitles,
                teacher
        );
    }

}

File: angular-ui/src/app/administration/courses/model/course.dto.ts
import {SectionDto} from "./section.dto";

export interface CourseDto {
  id: number;
  title: string;
  thumbnailUrl?: string;
  description?: string;
  language: string,
  subtitles?: string[],
  benefits?: string[],
  prerequisites?: string[],
  sections?: SectionDto[],
  price?: string,
  discountedPrice?: string,
  published: boolean,
  teacher: string,
  approvedBy?: string,
  // students: string[],
  discountCode?: string,
  createdBy: string,
  createdDate: string,
  lastModifiedBy: string,
  lastModifiedDate: string,
}
